### ISSUE TITLE
All templates for admin password reset are broken. What's wrong?

### ISSUE THREAD

**Issue by gremo** [sonata-project/SonataUserBundle#982]
How this is even possible? Why such big mistakes for a bundle used by thousands? The error is so obvious that it seems that this hasn't be tested at all.

I'm working on a PR to fix this.

#### Sonata packages

```
$ composer show --latest 'sonata-project/*'
sonata-project/admin-bundle              3.29.0 3.29.0 The missing Symfony Admin Generator
sonata-project/block-bundle              3.9.0  3.9.0  Symfony SonataBlockBundle
sonata-project/cache                     1.1.1  2.0.1  Cache library
sonata-project/cache-bundle              2.4.0  2.4.0  This bundle provides caching services
sonata-project/classification-bundle     3.5.0  3.5.0  Symfony SonataClassificationBundle
sonata-project/core-bundle               3.8.0  3.8.0  Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.3.1  2.3.1  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.0.2  1.0.2  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.3.0  3.3.0  Symfony Sonata / Integrate Doctrine ORM into the SonataAdmin...
sonata-project/easy-extends-bundle       2.3.0  2.3.0  Symfony SonataEasyExtendsBundle
sonata-project/exporter                  1.8.0  1.8.0  Lightweight Exporter library
sonata-project/formatter-bundle          3.4.0  3.4.0  Symfony SonataFormatterBundle
sonata-project/intl-bundle               2.4.0  2.4.0  Symfony SonataIntlBundle
sonata-project/media-bundle              3.10.0 3.10.0 Symfony SonataMediaBundle
sonata-project/notification-bundle       3.3.0  3.3.0  Symfony SonataNotificationBundle
sonata-project/page-bundle               3.7.0  3.7.0  This bundle provides a Site and Page management through cont...
sonata-project/seo-bundle                2.4.0  2.4.0  Symfony SonataSeoBundle
sonata-project/user-bundle               4.0.0  4.0.0  Symfony SonataUserBundle
```

#### Symfony packages

```
$ composer show --latest 'symfony/*'
symfony/assetic-bundle     v2.8.2  v2.8.2 Integrates Assetic into Symfony2
symfony/monolog-bundle     v3.1.2  v3.1.2 Symfony MonologBundle
symfony/phpunit-bridge     v3.4.2  v4.0.2 Symfony PHPUnit Bridge
symfony/polyfill-apcu      v1.6.0  v1.6.0 Symfony polyfill backporting apcu_* functions to lower PHP versions
symfony/polyfill-intl-icu  v1.6.0  v1.6.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring  v1.6.0  v1.6.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php56     v1.6.0  v1.6.0 Symfony polyfill backporting some PHP 5.6+ features to lower PHP versions
symfony/polyfill-php70     v1.6.0  v1.6.0 Symfony polyfill backporting some PHP 7.0+ features to lower PHP versions
symfony/polyfill-util      v1.6.0  v1.6.0 Symfony utilities for portability of PHP codes
symfony/security-acl       v3.0.0  v3.0.0 Symfony Security Component - ACL (Access Control List)
symfony/swiftmailer-bundle v2.6.7  v3.1.6 Symfony SwiftmailerBundle
symfony/symfony            v3.3.14 v3.4.2 The Symfony PHP framework
```

#### PHP version

```
$ php -v
PHP 7.1.12 (cli) (built: Nov 23 2017 04:33:43) ( NTS MSVC14 (Visual C++ 2015) x86 )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.1.12, Copyright (c) 1999-2017, by Zend Technologies
```

## Subject
Admin resetting complete broken.

## Steps to reproduce
Fresh install, try to reset your admin password (http://localhost:8000/admin/password-reset/request).

## Expected results
The form for input email used to restore is displayed.

## Actual results
Exception thrown:

> Variable "adminPool" does not exist.
> in vendor\sonata-project\user-bundle\src\Resources\views\Admin\Security\Resetting\request.html.twig (line 31)

**Comment by greg0ire** [comment]
> How this is even possible? Why such big mistakes for a bundle used by thousands? The error is so obvious that it seems that this hasn't be tested at all.

Either that or maybe Twig has become less permissive? I know there is some magic that will try several accessors for an object, maybe there was something similar for variables. Hard to believe it never worked / never was tested. If you really want to understand, you can checkout older versions and bisect from there.

**Comment by gremo** [comment]
@greg0ire I'm curious too,  but I don't think that Twig has been such permissive in the past about variable names. In fact, it was the 4.0 that changed the names in templates.

**Comment by OskarStark** [comment]
@gremo we released 4.0.1 right now

**Comment by gremo** [comment]
@OskarStark thank you so much!
